ignite/src/main/java/org/hibernate/ogm/datastore/ignite/query/parsing/impl/IgniteQueryRendererDelegate.java

```diff
@@ -0,0 +1,242 @@
+import org.hibernate.hql.ast.spi.EntityNamesResolver;
+import org.hibernate.hql.ast.spi.SingleEntityQueryBuilder;
+import org.hibernate.hql.ast.spi.SingleEntityQueryRendererDelegate;
+import org.hibernate.ogm.datastore.ignite.query.impl.IgniteQueryDescriptor;
+import org.hibernate.ogm.datastore.ignite.query.parsing.predicate.impl.IgnitePredicateFactory;
+import org.hibernate.ogm.persister.impl.OgmEntityPersister;
+
+/**
+ * Parser delegate which creates Ignite SQL queries in form of {@link StringBuilder}s.
+ *
+ * @author Victor Kadachigov
+ */
+public class IgniteQueryRendererDelegate extends SingleEntityQueryRendererDelegate<StringBuilder, IgniteQueryParsingResult> {
+
+	private static final List<String> ENTITY_COLUMN_NAMES = Collections.unmodifiableList( Arrays.asList( "_KEY", "_VALUE" ) );
+
+	private final IgnitePropertyHelper propertyHelper;
+	private final SessionFactoryImplementor sessionFactory;
+	private final Map<String, Object> namedParamsWithValues;
+	private List<Object> indexedParameters;
+	private List<OrderByClause> orderByExpressions;
+
+	public IgniteQueryRendererDelegate(SessionFactoryImplementor sessionFactory, IgnitePropertyHelper propertyHelper, EntityNamesResolver entityNamesResolver, Map<String, Object> namedParameters) {
+		super(
+				propertyHelper,
+				entityNamesResolver,
+				SingleEntityQueryBuilder.getInstance( new IgnitePredicateFactory( propertyHelper ), propertyHelper ),
+				namedParameters != null ? NamedParametersMap.INSTANCE : null /* we put '?' in query instead of parameter value */
+		);
+		this.propertyHelper = propertyHelper;
+		this.sessionFactory = sessionFactory;
+		this.namedParamsWithValues = namedParameters;
+	}
+
+	@Override
+	public void setPropertyPath(PropertyPath propertyPath) {
+		this.propertyPath = propertyPath;
+	}
+
+	private void where( StringBuilder queryBuilder ) {
+		StringBuilder where = builder.build();
+		if ( where != null && where.length() > 0 ) {
+			queryBuilder.append( " WHERE " ).append( where );
+		}
+	}
+
+	private void orderBy(StringBuilder queryBuilder) {
+		if ( orderByExpressions != null && !orderByExpressions.isEmpty() ) {
+			queryBuilder.append( " ORDER BY " );
+			int counter = 1;
+			for ( OrderByClause orderBy : orderByExpressions ) {
+				orderBy.asString( queryBuilder );
+				if ( counter++ < orderByExpressions.size() ) {
+					queryBuilder.append( ", " );
+				}
+			}
+		}
+	}
+
+	private void select(StringBuilder queryBuilder) {
+		queryBuilder.append( "SELECT _KEY, _VAL " );
+	}
+
+	private void from(StringBuilder queryBuilder) {
+		String tableAlias = propertyHelper.findAliasForType( targetTypeName );
+		OgmEntityPersister persister = (OgmEntityPersister) ( sessionFactory ).getEntityPersister( targetType.getName() );
+		String tableName = propertyHelper.getKeyMetaData( targetType.getName() ).getTable();
+		queryBuilder.append( " FROM " ).append( tableName ).append( ' ' ).append( tableAlias ).append( ' ' );
```

Comment:
Is that `tableName` also the name of the cache? If this is the case, you can  keep track of the value in the IgniteQueryDescriptor and use it in the dialect when `EntityMetadataInformation` is null (I think, I haven't test it).